### PR TITLE
Remove script for downloading debug adapters

### PR DESCRIPTION
**Pull request by msujew**
#### What it does

Closes https://github.com/eclipse-theia/theia/issues/8713

The script in question seems to be unused, as it was used to download debug adapters for the [`java-debug`](https://github.com/eclipse-theia/theia/pull/3613) package which was removed a few years ago. The script itself doesn't do a lot and can be easily replicated on user site using the `RequestService` if necessary.

#### How to test

There's nothing to test really, it mostly cleans up a few dependencies which were purely used inside of this script.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by vince-fugnitto** [comment]
I believe it fixes https://github.com/eclipse-theia/theia/issues/8713 :)

**Comment by msujew** [comment]
@vince-fugnitto Indeed :)

**Review by vince-fugnitto — APPROVED** [review]
LGTM :+1:

File: package.json [hunk 1]
```diff
@@ -44,6 +44,7 @@
     "node-gyp": "^9.0.0",
     "nsfw": "^2.2.4",
     "nyc": "^15.0.0",
+    "mkdirp": "^0.5.0",
```
Comment (paul-marechal) [inline package.json:47]: Why is this required here?

Comment (msujew) [reply package.json:47]: It's used in one of the performance scripts. See:

https://github.com/eclipse-theia/theia/blob/423b14b70b68826ddc8c8fe36e5afa7df5aae641/scripts/performance/extension-impact.js#L21

Comment (paul-marechal) [reply package.json:47]: The joy of dealing with implicit dependencies! Thanks for the explanation.
